examples/server/utils.hpp

```diff
@@ -133,26 +134,50 @@ struct completion_token_output
     std::string text_to_send;
 };
 
-static inline void server_log(const char *level, const char *function, int line,
-                       const char *message, const nlohmann::ordered_json &extra)
+static void server_log(const char *level, const char *function, int line, const char *message, const nlohmann::ordered_json &extra)
 {
-    nlohmann::ordered_json log
-    {
+    std::stringstream ss_tid;
+    ss_tid << std::this_thread::get_id();
+    json log = nlohmann::ordered_json{
+        {"tid", ss_tid.str()},
         {"timestamp", time(nullptr)},
-        {"level",     level},
-        {"function",  function},
-        {"line",      line},
-        {"message",   message},
     };
 
-    if (!extra.empty())
-    {
-        log.merge_patch(extra);
-    }
+    if (server_log_json) {
+        log.merge_patch(
+                {
+                        {"level",     level},
+                        {"function",  function},
+                        {"line",      line},
+                        {"msg",       message},
+                });
+        if (!extra.empty()) {
+            log.merge_patch(extra);
+        }
+
+        const std::string str = log.dump(-1, ' ', false, json::error_handler_t::replace);
+        printf("%.*s\n", (int)str.size(), str.data());
+        fflush(stdout);
+    } else {
+        char buf[1024];
+        snprintf(buf, 1024, "%24s %4s: %-80s", function, level, message);
 
-    const std::string str = log.dump(-1, ' ', false, json::error_handler_t::replace);
-    printf("%.*s\n", (int)str.size(), str.data());
-    fflush(stdout);
+        if (!extra.empty()) {
+            log.merge_patch(extra);
+        }
+        std::stringstream ss;
+        ss << buf;
+        for (const auto& el : log.items())
+        {
+            const std::string value = el.value().dump(-1, ' ', false, json::error_handler_t::replace);
+            snprintf(buf, 1024, " %s=%s", el.key().c_str(), value.c_str());
+            ss << buf;
```

Comment:
Here you can simply do `ss << el.key() << "=" << el.value();`

No need to explicitly convert json to string, since the compiler auto cast json into string when you concatenate it with another string.

Reply:
I was wondering what happen if the JSON serialization raised an error. I agree mixing both streams and snprintf is not perfect, but there is no point in the console text format to see lines > 1024 chars.